### PR TITLE
Add region to Get/DeleteBucket

### PR DESCRIPTION
**Pull request by tdmanv**
## Change Overview

We cannot discover the region for buckets in AWS S3 that use non-standard regions like `us-gov-west-1`. This PR adds region to Get/Delete

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

Ran a manual test with a bucket in `us-gov-west-1`:

https://gist.github.com/844c65e12f4ee9484522292f51c53807

- [x] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by DeepikaDixit** [comment]
Added hold off merging to stop auto merge. @tdmanv I accidentally approved instead of comment 🤦🏻‍♀️

**Comment by ntolia** [comment]
LGTM.

File: pkg/objectstore/bucket_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	c.Assert(err, NotNil)
+
+	err = p.DeleteBucket(ctx, bn, r)
+	c.Assert(err, ErrorMatches, ahmRe)
+	c.Assert(err, NotNil)
+
+	err = p.DeleteBucket(ctx, bn, r)
+	c.Assert(err, ErrorMatches, ahmRe)
+	c.Assert(err, NotNil)
+
```
Comment (ntolia) [inline pkg/objectstore/bucket_test.go:53]:
```suggestion
```

**Comment by tdmanv** [comment]
I leave the hold-off-merging tag on until I've put together the PR on the K10 side

**Pull request closed without merging**